pkg/etcd/etcd.go

```diff
@@ -47,15 +58,20 @@ func NewETCD() *ETCD {
 	}
 }
 
+var learnerProgressKey = version.Program + "/etcd/learnerProgress"
+
 const (
-	snapshotPrefix = "etcd-snapshot-"
-	endpoint       = "https://127.0.0.1:2379"
-	testTimeout    = time.Second * 10
-
-	// defaults from etcdctl/ctlv3/ctl.go
-	defaultDialTimeout      = 2 * time.Second
-	defaultKeepAliveTime    = 2 * time.Second
-	defaultKeepAliveTimeOut = 6 * time.Second
+	snapshotPrefix      = "etcd-snapshot-"
+	endpoint            = "https://127.0.0.1:2379"
+	testTimeout         = time.Second * 10
+	manageTickerTime    = time.Second * 15
```

Comment:
is there a reason why you used 15 seconds for each tick?

Reply:
It's somewhat arbitrary - wanted an interval that was long enough that it could be done with a deadline'd context that allowed for both slow responses due to load and connection timeouts due to nodes being offline, but frequent enough that it wouldn't cause significant delays in joining new members to the cluster. I'm open to suggestions.